{"version":3,"sources":["throw5.es6"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;AAAA;;;AAGA,QAAI,iCAAM,SAAU,GAAV;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA,+BAEI,QAAQ,GAAR,CAAY,GAAZ,CAFJ;;AAAA;AAAA;AAAA;;AAAA;AAAA;AAAA;;AAAA;AAAA;AAAA,+BAMA,QAAQ,GAAR,CAAY,GAAZ,CANA;;AAAA;AAAA;AAAA,+BAOA,QAAQ,GAAR,CAAY,GAAZ,CAPA;;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA,WAAU,GAAV;AAAA,KAAN,CAAJ;;AAUA,QAAI,IAAI,KAAR;AACA,MAAE,IAAF,G,CAAS;AACT,MAAE,KAAF,G,CAAU;AACV,MAAE,IAAF,G,CAAS","file":"throw5.js","sourcesContent":["/**\r\n * throw方法被捕获以后，会附带执行下一条yield语句。也就是说，会附带执行一次next方法。\r\n */\r\nvar gen = function* gen(){\r\n    try {\r\n        yield console.log('a');\r\n    } catch (e) {\r\n        // ...\r\n    }\r\n    yield console.log('b');\r\n    yield console.log('c');\r\n}\r\n\r\nvar g = gen();\r\ng.next() // a\r\ng.throw() // b\r\ng.next() // c"]}